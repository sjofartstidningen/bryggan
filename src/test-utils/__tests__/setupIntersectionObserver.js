import { setupIntersectionObserver } from '../';

describe('testUtil.IntersectionObserver', () => {
  let resetIntersectionObserver;
  beforeAll(() => {
    resetIntersectionObserver = setupIntersectionObserver();
  });

  afterAll(() => {
    resetIntersectionObserver();
  });

  test('should mock the IntersectionObserver API', () => {
    const observerCallback = jest.fn();
    const image = document.createElement('img');
    const button = document.createElement('button');

    const observer = new IntersectionObserver(observerCallback);
    observer.observe(image);
    observer.simulateIntersectionEvent({ isIntersecting: true });

    expect(observerCallback).toHaveBeenCalledTimes(1);
    expect(observerCallback.mock.calls[0]).toMatchSnapshot();

    observer.observe(button);
    observer.simulateIntersectionEvent({ isIntersecting: true });

    expect(observerCallback).toHaveBeenCalledTimes(2);
    expect(observerCallback.mock.calls).toMatchSnapshot();

    observer.unobserve(image);
    observer.simulateIntersectionEvent({ isIntersecting: true });

    expect(observerCallback).toHaveBeenCalledTimes(3);
    expect(observerCallback.mock.calls).toMatchSnapshot();
  });
});
