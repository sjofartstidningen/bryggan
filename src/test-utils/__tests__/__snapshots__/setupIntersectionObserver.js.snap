// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testUtil.IntersectionObserver should mock the IntersectionObserver API 1`] = `
Array [
  Array [
    Object {
      "isIntersecting": true,
      "target": <img />,
    },
  ],
]
`;

exports[`testUtil.IntersectionObserver should mock the IntersectionObserver API 2`] = `
Array [
  Array [
    Array [
      Object {
        "isIntersecting": true,
        "target": <img />,
      },
    ],
  ],
  Array [
    Array [
      Object {
        "isIntersecting": true,
        "target": <img />,
      },
      Object {
        "isIntersecting": true,
        "target": <button />,
      },
    ],
  ],
]
`;

exports[`testUtil.IntersectionObserver should mock the IntersectionObserver API 3`] = `
Array [
  Array [
    Array [
      Object {
        "isIntersecting": true,
        "target": <img />,
      },
    ],
  ],
  Array [
    Array [
      Object {
        "isIntersecting": true,
        "target": <img />,
      },
      Object {
        "isIntersecting": true,
        "target": <button />,
      },
    ],
  ],
  Array [
    Array [
      Object {
        "isIntersecting": true,
        "target": <button />,
      },
    ],
  ],
]
`;
